sdlib/d/gc/extent.d

```diff
@@ -266,7 +277,36 @@ public:
 		// FIXME: in contract.
 		assert(isSlab(), "slabData accessed on non slab!");
 
-		return _metadata.slabData;
+		return _metadata.bitmaps.slabData;
+	}
+
+	/**
+	 * Freespace Flag features for slabs.
+	 */
+
+	@property
+	ref shared(Bitmap!256) freeSpaceFlags() {
+		// FIXME: in contract.
+		assert(isSlab(), "slabData accessed on non slab!");
+
+		return _metadata.bitmaps.freeSpaceData.flags;
+	}
+
+	@property
+	bool allowsFreeSpace() const {
+		return isSlab() && isAppendableSizeClass(sizeClass);
+	}
+
+	bool hasFreeSpace(uint index) {
+		return allowsFreeSpace && freeSpaceFlags.valueAtAtomic(index);
+	}
+
+	void clearFreeSpace(uint index) {
+		freeSpaceFlags.setBitValueAtomic!false(index);
+	}
+
+	void setFreeSpace(uint index) {
+		freeSpaceFlags.setBitValueAtomic!true(index);
 	}
```

Comment:
The multiplication of accessor here is a bit unfortunate, especially since none of them gives us what we actually care about: how free space is there in that slot?